### PR TITLE
[Enhancement] - Optimize decay

### PR DESCRIPTION
**Pull request by beats-dh**
Co-Authored-By: Jakub @SaiyansKing jakubkubina@hotmail.com
From: https://github.com/opentibiabr/optimized_forgottenserver/tree/master

Reworked the decay, so the problems with the CPU spike are over, as well as other problems with the decay, like some items resetting the duration when equip/deequip.

New functions:
Game::stopDecay(item)
Container::stopDecaying()
DepotChest::isRemoved()
DepotLocker::isRemoved()
Inbox::isRemoved()
Item::stopDecaying()

From function:
decreaseDuration(time)
To:
setDurationTimestamp(timestamp)

New file decay.cpp/h
Decay::startDecay(item, duration)
Decay::stopDecay(item, timestamp)
Decay::checkDecay()

New attribute: ITEM_ATTRIBUTE_DURATION_TIMESTAMP

### CODE REVIEW

**Review by dudantas — APPROVED** [review]
Tested, everything seems to work as it should, we just need to know if the cpu usage will normalize, unfortunately it's a test I won't be able to do. But, in my view the merge is ready to be done, as the important thing is that everything is working, it is almost 100% certain that the cpu usage will no longer go up.

File: src/game/game.cpp [hunk 1]
```diff
@@ -1771,13 +1765,15 @@ ReturnValue Game::internalMoveItem(Cylinder* fromCylinder, Cylinder* toCylinder,
...
 		}
+		moveItem->startDecaying();
 	}
 
 	if (updateItem) {
 		int32_t updateItemIndex = toCylinder->getThingIndex(updateItem);
 		if (updateItemIndex != -1) {
 			toCylinder->postAddNotification(updateItem, fromCylinder, updateItemIndex);
 		}
+		moveItem->startDecaying();
```
Comment (dudantas) [inline src/game/game.cpp]:
```suggestion
		updateItem->startDecaying();
```

File: src/io/iologindata.cpp [hunk 2]
```diff
@@ -636,9 +638,10 @@ bool IOLoginData::loadPlayer(Player* player, DBResult_ptr result)
       int32_t pid = pair.second;
       if (pid >= 0 && pid < 100) {
         DepotChest* depotChest = player->getDepotChest(pid, true);
-        if (depotChest) {
-          depotChest->internalAddThing(item);
-        }
+				if (depotChest) {
```
Comment (dudantas) [inline src/io/iologindata.cpp]: Fix indentation, please

File: src/lua/scripts/luascript.cpp [hunk 3]
```diff
@@ -1590,7 +1590,7 @@ void LuaScriptInterface::registerFunctions()
 	registerEnum(ITEM_ATTRIBUTE_DOORID)
 	registerEnum(ITEM_ATTRIBUTE_SPECIAL)
 	registerEnum(ITEM_ATTRIBUTE_OPENCONTAINER)
-	registerEnum(ITEM_ATTRIBUTE_QUICKLOOTCONTAINER)
+	registerEnum(ITEM_ATTRIBUTE_DURATION_TIMESTAMP)
```
Comment (dudantas) [inline src/lua/scripts/luascript.cpp:1594]: Are you sure the replacement is correct?

**Comment by gerotib** [comment]
Only issue I see is that magic light wand timer is getting renewed each time we turn it off. I dont know if theres another item like this. (probably torches etc)
https://gfycat.com/linedtintedhoopoe

Before this pr it was also like this.

**Comment by dudantas** [comment]
> Only issue I see is that magic light wand timer is getting renewed each time we turn it off. I dont know if theres another item like this. (probably torches etc) https://gfycat.com/linedtintedhoopoe
> 
> Before this pr it was also like this.

Fixed.

**Review by gerotib — APPROVED** [review]

**Pull request merged**